### PR TITLE
B-27963: Util to Include Headers into outgoing context

### PR DESCRIPTION
**Pull request by sstaratsitarau**

### CODE REVIEW

File: util/context.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	if rid, ok := requestid.FromContext(ctx); ok {
+		params[requestid.DefaultRequestIDKey] = []string{rid}
+	}
+
+	if rIPs, ok := gateway.Header(ctx, gateway.XForwardedFor); ok {
+		params[gateway.XForwardedFor] = []string{rIPs}
+	}
+
+	return ForwardContextWithCustomParams(ctx, params)
+}
```
Comment (kd7lxl) [inline util/context.go:29]: Do tracing span-ids need to be forwarded here?

File: util/context.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+// ForwardContextWithCustomParams set to outgoing context all params from incoming context + custom params
+func ForwardContextWithCustomParams(ctx context.Context, params map[string][]string) context.Context {
```
Comment (kd7lxl) [inline util/context.go:32]: I'm not sure this function is needed. Usage of [google.golang.org/grpc/metadata](https://godoc.org/google.golang.org/grpc/metadata) is nearly as simple:
```go
md, _ := metadata.FromIncomingContext(ctx)
md = md.Join(params)
outgoingCtx := metadata.NewOutgoingContext(context.Background(), md)
```

File: gateway/header.go [hunk 3]
```diff
@@ -12,6 +12,11 @@ import (
 	"github.com/grpc-ecosystem/grpc-gateway/runtime"
 )
 
+const (
+	// XForwardedFor is a header to identify the origin of an IP address
+	XForwardedFor = "X-Forwarded-For"
+)
```
Comment (burov) [inline gateway/header.go:18]: What the reason to keep this constant here ?

File: util/context.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	if !ok {
+		return ctx
+	}
+
+	for key := range params {
+		md[key] = params[key]
+	}
+
+	return metadata.NewOutgoingContext(ctx, md)
+}
```
Comment (burov) [inline util/context.go:54]: I don't think we really need this function

File: util/context.go [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+	"github.com/infobloxopen/atlas-app-toolkit/requestid"
+)
+
+// ForwardContext set to outgoing context request_id, auth_token and X-Forwarded-For header value
+func ForwardContext(ctx context.Context) context.Context {
+	keys := []string{
+		auth.AuthorizationHeader,
+		requestid.DefaultRequestIDKey,
+		gateway.XForwardedFor,
+	}
```
Comment (burov) [inline util/context.go:19]: Probably add tracing context header will be good as well, see https://github.com/census-instrumentation/opencensus-go/blob/v0.22.4/trace/trace.go#L113 for more information

File: gateway/header.go (hunk 3 above)
Comment (sstaratsitarau) [reply gateway/header.go:18]: This is the key for the "X-Forwarded-For" value, which is the http request header. I figured the gateway package and header.go file would the most suitable place for this variable.

Could you suggest a better location for this variable?

File: util/context.go [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+	return ForwardContextWithCustomParams(ctx, keys)
+}
+
+// ForwardContextWithCustomParams set to outgoing context specified parameters from incoming context by keys
+func ForwardContextWithCustomParams(ctx context.Context, keys []string) context.Context {
```
[the remark this answers is not in the document]
Comment (sstaratsitarau) [reply util/context.go:25]: I don't quite understand why. We just pass keys, which are string values. For what reason should this way be changed to variadic parameters?

File: util/context.go (hunk 4 above)
Comment (Evgeniy-L) [reply util/context.go:54]: let remove this function
we can consider adding it if requested later

**Comment by addudko** [comment]
Closed due to 
https://github.com/infobloxopen/atlas-app-toolkit/pull/211

**Pull request closed without merging**